docs/source/models.rst

```diff
@@ -189,17 +189,86 @@ documentation <custom-python-models>`.
 
 How To Load And Score Python Function Models
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-You can load ``python_function`` models in Python by calling the :py:func:`mlflow.pyfunc.load_model()`
-function. Note that the ``load_model`` function assumes that all dependencies are already available
-and *will not* check nor install any dependencies (
-see :ref:`model deployment section <built-in-deployment>` for tools to deploy models with
-automatic dependency management).
 
-Once loaded, you can score the model by calling the :py:func:`predict <mlflow.pyfunc.PyFuncModel.predict>`
-method, which has the following signature::
+Loading Models
+##############
 
-  predict(data: Union[pandas.(Series | DataFrame), numpy.ndarray, csc_matrix, csr_matrix, List[Any], Dict[str, Any], str],
-          params: Optional[Dict[str, Any]] = None) → Union[pandas.(Series | DataFrame), numpy.ndarray, list, str]
+You can load ``python_function`` models in Python by using the :py:func:`mlflow.pyfunc.load_model()` function. It is important 
+to note that ``load_model`` assumes all dependencies are already available and *will not* perform any checks or installations 
+of dependencies. For deployment options that handle dependencies, refer to the :ref:`model deployment section <built-in-deployment>`.
+
+Scoring Models
+##############
+
+Once a model is loaded, it can be scored in two primary ways:
+
+1. **Synchronous Scoring**
+   The standard method for scoring is using the :py:func:`predict <mlflow.pyfunc.PyFuncModel.predict>` method, which supports various
+   input types and returns a scalar or collection based on the input data. The method signature is::
+
+        predict(data: Union[pandas.Series, pandas.DataFrame, numpy.ndarray, csc_matrix, csr_matrix, List[Any], Dict[str, Any], str],
+                params: Optional[Dict[str, Any]] = None) → Union[pandas.Series, pandas.DataFrame, numpy.ndarray, list, str]
+
+2. **Asynchronous Streaming Scoring**
```

Comment:
I'd avoid using asynchronous because we only support synchronous streaming, not async streaming.

Reply:
Good call